### PR TITLE
Remove most isPlainObject checks

### PR DESCRIPTION
**Pull request by evanbrooks-codaio**
As discussed here: https://staging.coda.io/d/_dTgmTD_HYPt#Prototype-Versions_tuUak/r21&view=modal

Various APIs use an `isPlainObject` check (`isText`, `isElement`, `isRange`, `isPoint`, etc). This appears to be a hot path. In our app we always pass in plain objects, and can trust TS a bit more than the library can. This change improves render time by around 2-3%.

I replaced most of these with truthy/existence checks, because it seems like there is code that relies on that.

WIP until I can get a proper patch generated in the main repo.

### CODE REVIEW

File: packages/slate-hyperscript/src/creators.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ const resolveDescendants = (children: any[]): Descendant[] => {
       const c = child // HACK: fix typescript complaining
 
       if (
+        prev &&
```
Comment (evanbrooks-codaio) [inline packages/slate-hyperscript/src/creators.ts:39]: Not strictly necessary as we don't use this module, just keeping the test passing

File: packages/slate/src/interfaces/editor.ts [hunk 2]
```diff
@@ -603,7 +603,7 @@ export const Editor: EditorInterface = {
       return cachedIsEditor
     }
 
-    if (!isPlainObject(value)) {
+    if (typeof value !== 'object') {
```
Comment (evanbrooks-codaio) [inline packages/slate/src/interfaces/editor.ts:606]: I kept this simpler check here, because we try to use the value as a WeakMap key on line 631, which will crash if a primitive value slips through. (This may only be happening in tests, can remove if seems safe?)

File: packages/slate/src/interfaces/point.ts [hunk 3]
```diff
@@ -85,10 +84,8 @@ export const Point: PointInterface = {
    */
 
   isPoint(value: any): value is Point {
-    return (
-      isPlainObject(value) &&
-      typeof value.offset === 'number' &&
-      Path.isPath(value.path)
+    return Boolean(
+      value && typeof value.offset === 'number' && Path.isPath(value.path)
```
Comment (evanbrooks-codaio) [inline packages/slate/src/interfaces/point.ts:88]: I'm checking for existence here because `isRange` will pass in potentially-undefined points. Should I check for existence everywhere?

File: packages/slate/src/interfaces/editor.ts (hunk 2 above)
Comment (steve-codaio) [reply packages/slate/src/interfaces/editor.ts:606]: It's probably safe, though this one typeof compared to the myriad down below doesn't seem like it'll gain much for us

File: packages/slate/test/index.js [hunk 4]
```diff
@@ -2,11 +2,12 @@ import assert from 'assert'
 import { fixtures } from '../../../support/fixtures'
 import { Editor } from 'slate'
 import { createHyperscript } from 'slate-hyperscript'
+import { isPlainObject } from 'is-plain-object'
 
 describe('slate', () => {
   fixtures(__dirname, 'interfaces', ({ module }) => {
     let { input, test, output } = module
-    if (Editor.isEditor(input)) {
+    if (isPlainObject(input) && Editor.isEditor(input)) {
```
Comment (steve-codaio) [inline packages/slate/test/index.js:10]: why did you need to add this?

**Review by steve-codaio — APPROVED** [review]

File: packages/slate/test/index.js (hunk 4 above)
Comment (evanbrooks-codaio) [reply packages/slate/test/index.js:10]: The test fixture passes in all kind of random input to `Editor.isEditor`, which caused the tests crash due to input used as WeakMap key. but i think i may have fixed it via the other change in editor.ts, let me try and remove

**Comment by evanbrooks-codaio** [comment]
Forgot this was still open. Had hit issues getting the patch to apply properly and moved on, can take another pass if we think its worth it
